src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/FeatureSwitchManager.cs

```diff
@@ -142,6 +146,9 @@ public MethodIL GetMethodILWithInlinedSubstitutions(MethodIL method)
 
             Debug.Assert(method.GetMethodILDefinition() == method);
 
+            bool shouldInlineResourceStrings =
```

Comment:
Up until now, the compiler didn't need to be aware of `System.Resources.UseSystemResourceKeys`. It just fell out from the descriptors. Now it needs to be aware. I don't particularly like that but don't know what else to do.

Reply:
Personally I don't have a problem with the compiler knowing about certain feature switches - in particular AOT compiler is almost like "half of the runtime" anyway. I would use a different feature switch for this though - as it has different behavior from this one.